### PR TITLE
fix textarea newline handle

### PR DESCRIPTION
**Pull request by metiftikci**
- Fix cursor position if input newline on middle of lines
- ~Increment number if numbered list~

image

### CODE REVIEW

File: web_src/js/features/comp/EditorMarkdown.ts [hunk 1]
```diff
@@ -92,11 +92,12 @@ function handleNewline(textarea: HTMLTextAreaElement, e: Event) {
   if (!line) {
     // clear current line if we only have i.e. '1. ' and the user presses enter again to finish creating a list
     textarea.value = value.slice(0, lineStart) + value.slice(lineEnd);
+    textarea.setSelectionRange(selStart - prefix.length, selStart - prefix.length);
   } else {
     // start a new line with the same indention and prefix
     let newPrefix = prefix;
     // a simple approach, otherwise it needs to parse the lines after the current line
-    if (/^\d+\./.test(prefix)) newPrefix = `1. ${newPrefix.slice(newPrefix.indexOf('.') + 2)}`;
```
Comment (metiftikci) [inline web_src/js/features/comp/EditorMarkdown.ts]: ~not sure what old code does~

got it

Comment (wxiaoguang) [reply web_src/js/features/comp/EditorMarkdown.ts]: Comment above `    // a simple approach, otherwise it needs to parse the lines after the current line`

**Comment by metiftikci** [comment]
@wxiaoguang should i revert the increase number code or try to implement increasing numbers below?

**Comment by wxiaoguang** [comment]
> @wxiaoguang should i revert the increase number code or try to implement increasing numbers below?

Either is fine, but there are many edge cases, even GitHub doesn't get it completely right, and it would be quite complex, maybe at least hundreds lines of code in my mind.

So at the moment, I would prefer to revert "the increase number"

**Comment by metiftikci** [comment]
@wxiaoguang done

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> @wxiaoguang should i revert the increase number code or try to implement increasing numbers below?

I have made a complete fix :  Fix editor markdown not incrementing in a numbered list #33187

**Pull request merged**